tests/orquestra/opt/api/optimizer_contract_test.py

```diff
@@ -7,13 +7,51 @@
 import pytest
 
 from orquestra.opt.api import CostFunction
-from orquestra.opt.api.optimizer_test import NESTED_OPTIMIZER_CONTRACTS
+from orquestra.opt.api.optimizer_test import (
+    NESTED_OPTIMIZER_CONTRACTS,
+    _validate_gradients_history_is_recorded_if_keep_history_is_true,
+    _validate_optimizer_does_not_record_history_by_default,
+    _validate_optimizer_does_not_record_history_if_keep_history_is_false,
+    _validate_optimizer_records_history_if_keep_history_is_true,
+)
 from orquestra.opt.mock_objects import (
     MockNestedOptimizer,
     MockOptimizer,
     mock_cost_function,
 )
 
+# Because MockOptimizer is not a precise optimizer, it cannot be used to validate the
+# contracts that test for accuracy of optimization result values.
+CONTRACTS_THAT_DONT_TEST_ACCURACY = [
```

Comment:
Maybe we should add `accuracy` as an optional argument for all the contracts?
We ran into some issues with accuracy for backend tests recently, I think this might solve a lot of issues in future. WDYT?

Reply:
Yes, I mean sth like:
```
def _validate_optimizer_succeeds_with_optimizing_rosenbrock_function(optimizer, accuracy:Optional[float]=1e-4):
    cost_function = FunctionWithGradient(
        rosenbrock_function, finite_differences_gradient(rosenbrock_function)
    )

    results = optimizer.minimize(cost_function, initial_params=np.array([0, 0]))

    return (
        results.opt_value == pytest.approx(0, abs=accuracy)
        and results.opt_params == pytest.approx(np.ones(2), abs=accuracy*10)
        and all(field in results for field in MANDATORY_OPTIMIZATION_RESULT_FIELDS)
    )
```
True, for contracts which don't test for accuracy we don't need accuracy :)

Reply:
Done!